### PR TITLE
enable to reopen evaluation

### LINKED ISSUE #77: Possibility to change evaluation period during or after evaluation
It should be possible to re-open evaluations after the end of the evaluation period.

Likewise, it should be possible to change the period when the course is in evaluation.

### PR DESCRIPTION
**Pull request by janno42**
fix #77

based on #526

### CODE REVIEW

File: evap/staff/forms.py [hunk 1]
```diff
@@ -60,9 +62,12 @@ def __init__(self, *args, **kwargs):
...
         self.fields['last_modified_user_2'].widget.attrs['readonly'] = "True"
 
-        if self.instance.state == "inEvaluation":
+        if self.instance.state in ['inEvaluation', 'evaluated', 'reviewed']:
             self.fields['vote_start_date'].widget.attrs['readonly'] = "True"
-            self.fields['vote_end_date'].widget.attrs['readonly'] = "True"
+
+    def clean(self):
+        if self.cleaned_data['vote_end_date'] < datetime.date.today():
+            raise forms.ValidationError(_("The vote end date must be in the future."))
```
Comment (karyon) [inline evap/staff/forms.py]: was it possible previously to create courses which end date was not in the future?

Comment (janno42) [inline evap/staff/forms.py]: i guess so

**Comment by janno42** [comment]
i see the errors, will fix them tomorrow.

File: evap/evaluation/models.py [hunk 2]
```diff
@@ -248,6 +248,10 @@ def revert_to_new(self):
     def evaluation_begin(self):
         pass
 
+    @transition(field=state, source=['evaluated', 'reviewed'], target='inEvaluation')
```
Comment (karyon) [inline evap/evaluation/models.py]: maybe add a condition here? im not sure...

**Comment by janno42** [comment]
added condition, fixed tests

**Comment by karyon** [comment]
:+1:

**Pull request merged**